app/models/grda_warehouse/auth_policies/client_policy.rb

```diff
@@ -0,0 +1,142 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class GrdaWarehouse::AuthPolicies::ClientPolicy < GrdaWarehouse::AuthPolicies::BasePolicy
+  # expose role permissions. Optionally rename the permission
+  [
+    [:can_view_client_name, :can_view_name?],
+    [:can_view_client_photo, :can_view_photo?],
+    [:can_view_full_dob],
+    [:can_view_full_ssn],
+    [:can_view_hiv_status],
+  ].each do |permission, method_name|
+    method_name ||= :"#{permission}?"
+    define_method(method_name) do
+      permission_granted_by_role?(permission)
+    end
+    memoize method_name
+  end
+
+  # can the user see the full client dash page and additional details?
+  memoize def can_view?
+    return true if permission_granted_through_roi?(:can_view_client_enrollments_with_roi)
+
+    permission_granted_by_role?(:can_view_clients)
```

Comment:
For performance, I wonder if the order of these should be flipped.  Maybe checking ROI is faster since that tends to be a smaller dataset and fewer installations.